src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/mapper/PrisonMovementEventsMapper.kt

```diff
@@ -0,0 +1,72 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.service.mapper
+
+import org.mapstruct.Mapper
+import org.mapstruct.Mapping
+import org.mapstruct.ValueMapping
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.client.prisonapi.PrisonMovementEvent
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.client.prisonapi.PrisonMovementEvents
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.client.prisonapi.PrisonMovementType
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.resource.mapper.InstantMapper
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.timeline.TimelineEvent
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.timeline.TimelineEventType
+import java.time.Instant
+import java.util.UUID
+
+@Mapper(
+  uses = [
+    InstantMapper::class,
+  ],
+  imports = [
+    Instant::class,
+    UUID::class,
+  ],
+)
+abstract class PrisonMovementEventsMapper {
+  fun toTimelineEvents(prisonMovementEvents: PrisonMovementEvents): List<TimelineEvent> {
+    val timelineEvents = mutableListOf<TimelineEvent>()
+    prisonMovementEvents.prisonMovements.map {
```

Comment:
There's some inherent complexity here as we're converting a Map of `<Long, PrisonMovementEvent>` to  a flat List of `TimelineEvents`. Hopefully this is readable, but suggestions are welcomed 👍